juniper/src/executor/mod.rs

```diff
@@ -720,6 +699,41 @@ where
     Ok((value, errors))
 }
 
+pub fn get_operation<'a, 'b, S>(
```

Comment:
Thanks for pulling out the duplicated code to a function. Can we get a unit test for it?

Reply:
Ok, I will try to make some tests specific for this function.
However, this is not new functionality so as a whole it should be covered by existing test cases.

Reply:
Totally. Just before the invariants were tangentially tested due to the implementation around this code. Now that this is a function with discrete inputs and outputs we should have some test that ensure the input and output invariants hold on their own.

Reply:
I had a look at the test suite, I found that all the relevant cases for `get_operation` are covered in test module `named_operations` at `juniper/src/executor_tests/executor.rs:1061`.

Here are the test cases:

- `uses_inline_operation_if_no_name_provided`
- `uses_only_named_operation`
- `uses_named_operation_if_name_provided`
- `error_if_multiple_operations_provided_but_no_name`
- `error_if_unknown_operation_name_provided`

I was thinking of having unit test cases for `get_operation` very similar to these above.
However, in my experience, when a thorough testing of a specific feature is possible at the integration-test level without extra effort, I prefer having only integration tests and not unit tests (integration tests are more robust against refactorings, and also test more stuff with less code, etc.).
Since this seems the case here, IMHO adding unit tests here would not add any value, and instead make the code less maintainable.

Let me know your thought about this, and also if you see a better option.

Sorry for the long talk and no code :sweat_smile: 
If you think it is better to have the tests for whatever reason, I can definitely write them! :smiley: